dts/bindings/pinctrl/nordic,nrf-pinctrl.yaml

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    nRF pin controller
```

Comment:
I think the major things to cover here are:

- this is a singleton that is already provided for every SoC
- it has a node label named `pinctrl` you can use to set properties
- it should be changed in board files and applications to contain a global set of pin configurations for all peripherals
- example pin configurations (e.g. UART)

I know this is just a draft, and I'm not asking for this to be done right now. Just making a note for later.

Reply:
I've re-written the description, I think it now contains all the points you mention. Let me know if it's better now.